urlce7759b8da_Palachsuper_EBA4_tgz-pJ8-json_TestJ8/src/terminal/tools/PropTools.java

```diff
@@ -0,0 +1,115 @@
+package terminal.tools;
+
+import java.io.*;
+import java.util.Properties;
+
+public class PropTools {
+
+	 /**
+     * Load a Properties File
+     * @param propsFile
+     * @return Properties
+     * @throws java.io.IOException
+     */
+    public static Properties load(String propsFile)
+    {
+    	Properties props = new Properties();
+        FileInputStream fis;
+		try {
+			fis = new FileInputStream(new File(propsFile));
+			props.load(fis);    
+		    fis.close();
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+       
+        return props;
+    }
+    
+    public static String getPropFile()
+    {
+    	String propFile=""; 
+		propFile = "../service.properties";
+		 
+		return propFile;
+    }
+	
+    public static void setProperty(String propertyName, String propertyValue)
+    {
+    	String propFile = PropTools.getPropFile();
+    	Properties prop = PropTools.load(propFile);
+    	
+    	prop.setProperty(propertyName, propertyValue);
+    	
+    	try 
+    	{ 
+    		FileOutputStream sF=new FileOutputStream( propFile );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *sF*. Execution paths that do not contain closure statements were detected. Ensure *sF* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).